### PR TITLE
Rm/user settings get

### PR DESCRIPTION
**Pull request by rickmarry**
### :chains: Related Resources

#4953 

### :athletic_shoe: How to Build and Test the Change
 from within hab run the following:
 ```
 start_all_services
 rebuild components/automate-deployment
 rebuild components/automate-gateway
 rebuild components/user-settings-service
 ```

once all are rebuilt, you can run the following from within hab to see that we may retrieve the _default user settings

`user_settings_grpcurl_get_user_settings _default local`

response should be:
```
{
  "user": {
    "name": "_default",
    "connector": "local"
  },
  "settings": {
    "date_format": {
      "value": "EEEE, MMMM d, y, h:mm:ss a zzzz",
      "enabled": true
    }
  }
}
```

 the first argument is the user name, in this case: _default
 the second argument is the connector, in this case: local
 
 next, you may make this call to the service, using any user or connector and if they don't currently have any user settings that have been overridden, the response should be just those of the _default user
 
 try it with user: "some_non_existant_user" connector: "local"
 
 `user_settings_grpcurl_get_user_settings some_non_existant_user local`
 
 response should be:
 ```
 {
  "user": {
    "name": "some_non_existant_user",
    "connector": "local"
  },
  "settings": {
    "date_format": {
      "value": "EEEE, MMMM d, y, h:mm:ss a zzzz",
      "enabled": true
    }
  }
}
```

 
            
            

 
### :white_check_mark: Checklist

**All PRs** must tick these:

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [x] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [x] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [x] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [x] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [x] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [x] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### CODE REVIEW

File: api/config/deployment/automate_config.proto [hunk 1]
```diff
@@ -102,4 +103,5 @@ message AutomateConfig {
 	chef.automate.domain.infra_proxy.ConfigRequest infra_proxy = 45;
 	chef.automate.domain.cds.ConfigRequest cds = 46;
 	chef.automate.domain.sample_data.ConfigRequest sample_data = 47;
+  chef.automate.domain.user_settings.ConfigRequest user_settings = 48;
```
Comment (vivek-yadav) [inline api/config/deployment/automate_config.proto:106]: I think there is an alignment issue here.

Comment (rickmarry) [reply api/config/deployment/automate_config.proto]: good catch.. fixed

**Review by punitmundra — APPROVED** [review]
change look fine to me. 
I have build the changes and tested locally. 
Just add the export comment at few of the function, rest of the thing is ok

**Review by vivek-yadav — APPROVED** [review]

**Comment by rickmarry** [comment]
I added two small commits to this since it was approved: 
 https://github.com/chef/automate/pull/5121/commits/d9d6a84d18d510f943113c1c6a6ca4c06d90451f just added an array of valid values to each respective user setting
 
 https://github.com/chef/automate/pull/5121/commits/bcc491b4da9a8eedda430f2469472e866aa3fe27 puts comments on the exported funcs as per @punitmundra request

**Pull request merged**